### PR TITLE
[docs] Add information on virtualenv

### LINKED ISSUE #6565: Document virtualenv setup steps
Trying to run "./wptrun firefox payment-request" yields the following error: 

```Bash
./wptrun firefox payment-request                                                                              
Traceback (most recent call last):
  File "./wptrun", line 5, in <module>
    wptrun.main()
  File "/Users/marcos/temp/web-platform-tests/tools/wptrun.py", line 304, in main
    venv = virtualenv.Virtualenv(os.path.join(wpt_root, "_venv_%s") % platform.uname()[0])
  File "/Users/marcos/temp/web-platform-tests/tools/browserutils/virtualenv.py", line 15, in __init__
    raise ValueError("virtualenv must be installed and on the PATH")
ValueError: virtualenv must be installed and on the PATH
```

The README seems to be missing documentation for "virtualenv".

### PR DESCRIPTION
**Pull request by jugglinmike**
Many of the subcommands of the WPT CLI require the `virtualenv` utility,
but the installation instructions did not instruct contributors to
install it. Add an installation step describing how to install the
dependency.

---

This is intended to resolve gh-6565.

[I initially wrote this patch to assume Python 2.7.9](https://github.com/web-platform-tests/wpt/compare/master...bocoup:docs-virtualenv-279) since that would allow us to take the presence of Pip for granted. However, it seems like older (but still common) releases of macOS may bundle prior versions of Python. If that's the case, it seems better to add a potentially-unnecessary invocation of `easy_install` than to try to guide users on updating their system Python.

### CODE REVIEW

**Review by sideshowbarker — APPROVED** [review]

**Review by Ms2ger — APPROVED** [review]

File: docs/_running-tests/from-local-system.md [hunk 1]
```diff
@@ -14,6 +14,13 @@ On Windows, be sure to add the Python directory (`c:\python2x`, by default) to
 your `%Path%` [Environment Variable](http://www.computerhope.com/issues/ch000549.htm),
 and read the [Windows Notes](#windows-notes) section below.
 
+Install `virtualenv` using the following commands:
+
+```bash
+easy_install pip
```
Comment (gsnedders) [inline docs/_running-tests/from-local-system.md]: `pip` should be installed by default with at all recent Python versions (from 2.7.9, from Dec 2014)?

**Comment by jugglinmike** [comment]
> `pip` should be installed by default with at all recent Python versions (from 2.7.9, from Dec 2014)?

That's right. I originally wrote this to update the required Python version from "2.7+" to "2.7.9+". I changed course because I'm concerned about macOS. It's a popular OS among potential contributors, but I wasn't able to find information about which Python version is bundled in each release. I decided that an extra (potentially-superfluous) command would be preferable to additional steps about checking the version of the system Python and potentially upgrading it, 

If anyone can find some authoritative statement about the Python version available in recent macOS release, then I'd be glad to remove this.

**Comment by gsnedders** [comment]
> If anyone can find some authoritative statement about the Python version available in recent macOS release, then I'd be glad to remove this.

10.14.4 has 2.7.10 in `/usr/bin/python`. However, it doesn't seem to install a `/usr/bin/pip` (though there is `/Library/Python/2.7/site-packages/pip`, so maybe the module is installed by default?). So, uh, yuk. I'll need to check what's up on a clean install, and I can't do that later till later this week.

Also, https://pip.pypa.io/en/stable/installing/ seems to avoid `easy_install` nowadays.

**Comment by jugglinmike** [comment]
Just ran across an alternative usage:

    python -m pip install  --user virtualenv

This ought to solve the problem you described @gsnedders. What do you think?

**Comment by gsnedders** [comment]
So on 10.13, given this seems to be what my clean VM is, `pip` isn't installed by default *but* `ensurepip` is shipped (as one would expect). You can therefore install pip with `python -m ensurepip`.

*However*, this needs root permission because `/Library/Python/2.7` etc. is only writable by root. `python -m ensurepip --user` works as you'd expect, though `~/Library/Python/2.7/bin` isn't on the `PATH` by default. `~/Library/Python/2.7/bin/pip install --user virtualenv` will then give you a `~/Library/Python/2.7/bin/virtualenv` which should be usable.

So, tl;dr: I think we should recommend `ensurepip` or whatever https://pip.pypa.io/en/stable/installing/ says, though we do need to concern ourselves as to whether the default install path is user-writable.

**Comment by jugglinmike** [comment]
Thanks for the extra research, @gsnedders.

> So, tl;dr: I think we should recommend `ensurepip`

Unfortunately, that's not going to fly on Ubuntu:

    $ lsb_release -a
    No LSB modules are available.
    Distributor ID: Ubuntu
    Description:    Ubuntu 18.04.2 LTS
    Release:        18.04
    Codename:       bionic
    $ python -m ensurepip
    ensurepip is disabled in Debian/Ubuntu for the system python.
    
    Python modules For the system python are usually handled by dpkg and apt-get.
    
        apt-get install python-<module name>
    
    Install the python-pip package to use pip itself.  Using pip together
    with the system python might have unexpected results for any system installed
    module, so use it on your own risk, or make sure to only use it in virtual
    environments.

> or whatever https://pip.pypa.io/en/stable/installing/ says 

That page makes no mention of `ensurepip`, but it *does* have the following warning:

> **Warning:** Be cautious if you are using a Python install that is managed by your operating system or another package manager. `get-pip.py` does not coordinate with those tools, and may leave your system in an inconsistent state.

This is definitely the case for Ubuntu, and since Homebrew is so popular among macOS users, it seems like a source of confusion for them, too.

It's starting to sound like there is no single command sequence that will work for all contributors. Could you explain where/how using `easy_install` will fail?

**Comment by gsnedders** [comment]
This just makes it seem like we can't provide simple, universal installation instructions for pip. :/

**Comment by zcorpan** [comment]
Having something that says `virtualenv` is required is better than the status quo. Can we merge, and ponder about improvements as a follow-up?

**Comment by jugglinmike** [comment]
@zcorpan that sounds good to me. The main issue is about pip. I've rewritten that part to be more hand-wavy. Does that look good to you and @gsnedders?

Separately, the error reported by the CLI has become a little less helpful since gh-6565 was opened. I've submitted gh-17033 to address that.

I had to rebase to resolve conflicts resulting from the merge of gh-16822. [The original version of this branch is available here.](https://github.com/web-platform-tests/wpt/compare/master...bocoup:docs-virtualenv-orig)

**Review by zcorpan — APPROVED** [review]

**Review by gsnedders — APPROVED** [review]

**Comment by Hexcles** [comment]
Congrats! I only saw this PR today and reading all these discussions gives me goosebumps. Thanks for going down the rabbit hole and finding the best cross-platform approach!

**Comment by jugglinmike** [comment]
The "final" text is a compromise, so it's kind of unsatisfying. I especially appreciate your encouragement on this one :)
